UnicornEngine/video/source/vulkan/Image.cpp

```diff
@@ -0,0 +1,159 @@
+/*
+* Copyright (C) 2017 by Godlike
+* This code is licensed under the MIT license (MIT)
+* (http://opensource.org/licenses/MIT)
+*/
+
+#include <unicorn/video/vulkan/Image.hpp>
+#include <unicorn/utility/Logger.hpp>
+
+namespace unicorn
+{
+namespace video
+{
+namespace vulkan
+{
+Image::Image(): m_image(nullptr)
+              , m_format()
+              , m_width(0)
+              , m_height(0)
+              , m_initialized(false)
+{
+}
+
+Image::~Image()
+{
+    Destroy();
+}
+
+bool Image::IsInitialized() const
+{
+    return  m_initialized;
+}
+
+bool Image::Create(vk::PhysicalDevice physicalDevice, vk::Device device, vk::Format format, vk::ImageUsageFlags usage, uint32_t width, uint32_t height)
+{
+    m_device = device;
```

Comment:
If you're not checking the value of `m_initialized` before you're creating new one, it seems like some information will not be properly deinitialized.

Do you need a way to reuse the same object but with different contents or could you move the initialization part to constructor?
If you need to be able to change the content within the object - please check `m_initialized` and reset the object before proceeding with the rest of initialization

Also right now you're compromising the state of the object in case when first `Create` was successful and the other one wasn't because you're not resetting `m_initialized` and it will still hold `true` even though the image was not created.